src/app/shared/search-box/search-box.component.ts

```diff
@@ -127,18 +129,42 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
 
   public onAddQueryItem(queryItem: QueryItem) {
     this.queryItems.push(queryItem);
-    this.queryItemsControl.push(queryItemToForm(queryItem));
+
+    const queryItemForm = queryItemToForm(queryItem);
+    this.queryItemsControl.push(queryItemForm);
+
+    if (queryItemForm.valid) {
```

Comment:
If I understand it correctly, `queryItemForm` represents a form group responsible for just a single query item. What happens if you already have an invalid query item in the search box and add a new one which is valid? This condition will be true but the form should not be submitted.

Reply:
Good point! I will definitely change it :)